src/base/bittorrent/session.cpp

```diff
@@ -1935,7 +1935,17 @@ void Session::networkConfigurationChange(const QNetworkConfiguration& cfg)
     if (configuredInterfaceName.isEmpty()) return;
 
     const QString changedInterface = cfg.name();
+
+    // workaround for QTBUG-52633: check interface IPs, react only if the IPs have changed
+    // seems to be present only with NetworkManager, hence Q_OS_LINUX
+#if defined Q_OS_LINUX && QT_VERSION >= QT_VERSION_CHECK(5, 0, 0) // && QT_VERSION <= QT_VERSION_CHECK(5, ?, ?)
```

Comment:
I hope Qt devs will eventually fix this bug and the second (commented out now) condition will be activated. Also, I don't know whether the problem is present in all Qt 5 versions, so perhaps the first condition may be tuned.